server/events/vcs/github_client_test.go

```diff
@@ -591,6 +591,157 @@ func TestGithubClient_PullIsMergeable(t *testing.T) {
 	}
 }
 
+func TestGithubClient_PullIsMergeableWithAllowMergeableBypassApply(t *testing.T) {
```

Comment:
Just curious on your thoughts on creating a new test instead of directly changing `TestGithubClient_PullIsMergeable` (by adding `allowMergeableBypassApply` field in the test case struct maybe)? I feel either way is fine, but the latter might be a bit cleaner.

Reply:
Basically, I felt that it would be too many cases in the struct and it would be hard to understand them.

Reply:
Sounds good. Just that if we ever want to modify anything in `PullIsMergeable`, we might need to change both tests. (Of course it might not happen.)

Reply:
yes, or if functionality behind the feature flag becomes default and only behavior, it would be enough to remove old test :)